pkg/openfeature/client.go

```diff
@@ -563,6 +564,18 @@ func (c *Client) evaluate(
 		"evaluationContext", evalCtx, "evaluationOptions", options,
 	)
 
+	evalDetails := InterfaceEvaluationDetails{
+		Value: defaultValue,
+		EvaluationDetails: EvaluationDetails{
+			FlagKey:  flag,
+			FlagType: flagType,
+		},
+	}
+
+	if !utf8.Valid([]byte(flag)) {
+		return evalDetails, NewParseErrorResolutionError("flag key is not a UTF-8 encoded string")
+	}
```

Comment:
Besides flag key, is there any other places we should validate this?

Is it worth doing it for context keys and values, for instance (that could incur some performance penalties... Just a thought!

Reply:
Perhaps there is value to that, should we create a separate issue to raise the question?

Reply:
https://github.com/open-feature/go-sdk/issues/147